components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/RefreshGrantHandler.java

```diff
@@ -146,11 +148,14 @@ public boolean validateScope(OAuthTokenReqMessageContext tokReqMsgCtx)
          */
         String[] requestedScopes = tokReqMsgCtx.getOauth2AccessTokenReqDTO().getScope();
         String[] grantedScopes = tokReqMsgCtx.getScope();
+        String[] grantedInternalScopes = tokReqMsgCtx.getAuthorizedInternalScopes();
```

Comment:
From where does this get populated?

Reply:
From here: 
https://github.com/wso2-extensions/identity-inbound-auth-oauth/blob/5211ddffb34c55ca0371febea86b5cd43f72269d/components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/AccessTokenIssuer.java#L421